src/core/gui/menus/PageTypeSelectionMenuBase.h

```diff
@@ -0,0 +1,56 @@
+/*
+ * Xournal++
+ *
+ * Handles page selection menus -- Base
+ *
+ * @author Xournal++ Team
+ * https://github.com/xournalpp/xournalpp
+ *
+ * @license GNU GPLv2 or later
+ */
+
+#pragma once
+
+#include <optional>
+#include <string_view>
+
+#include <gio/gio.h>  // for GSimpleAction
+#include <glib.h>
+
+#include "model/PageType.h"  // for PageType
+#include "util/raii/GObjectSPtr.h"
+
+class PageTypeHandler;
+class PageTypeInfo;
+class Settings;
+
+class PageTypeSelectionMenuBase {
+public:
+    PageTypeSelectionMenuBase(PageTypeHandler* typesHandler, const Settings* settings,
+                              const std::string_view& actionName);
+    virtual ~PageTypeSelectionMenuBase() = default;
```

Comment:
It looks like, this class should never be destructed directly, make the destructor protected and non virtual.